editoast/src/views/v2/timetable.rs

```diff
@@ -240,6 +247,46 @@ async fn delete(
     Ok(HttpResponse::NoContent().finish())
 }
 
+/// Create train schedule by batch
+#[utoipa::path(
+    tag = "train_schedulev2",
+    params(TimetableIdParam),
+    request_body = Vec<TrainScheduleForm>,
+    responses(
+        (status = 200, description = "The created train schedules", body = Vec<TrainScheduleResult>)
+    )
+)]
+#[post("train_schedule")]
+async fn train_schedule(
+    db_pool: Data<DbPool>,
+    timetable_id: Path<TimetableIdParam>,
+    data: Json<Vec<TrainScheduleBase>>,
+) -> Result<Json<Vec<TrainScheduleResult>>> {
+    use crate::modelsv2::CreateBatch;
+
+    let conn = &mut db_pool.get().await?;
+
+    let timetable_id = timetable_id.id;
+    TimetableWithTrains::retrieve_or_fail(conn, timetable_id, || TimetableError::NotFound {
+        timetable_id,
+    })
+    .await?;
+
+    let changesets: Vec<TrainScheduleChangeset> = data
+        .into_inner()
+        .into_iter()
+        .map(|ts| TrainScheduleForm {
+            timetable_id: Some(timetable_id),
+            train_schedule: ts,
+        })
+        .map_into()
+        .collect();
```

Comment:
It's fine like this, just to note that create_batch* functions accept an `IntoIterator<Item = Changeset>`, so you can omit collect if you want.